File: node/app/models/AsistenteModel.js
const db = require('../database');

class AsistenteModel {
    async findAll() {
        const [rows] = await db.query(`
            SELECT a.*, e.nombre as evento_nombre 
            FROM asistentes a 
            LEFT JOIN eventos e ON a.evento_id = e.id
        `);
        return rows;
    }

    async findById(id) {
        const [rows] = await db.query('SELECT * FROM asistentes WHERE id = ?', [id]);
        return rows[0];
    }

    async findByEvento(evento_id) {
        const [rows] = await db.query('SELECT * FROM asistentes WHERE evento_id = ?', [evento_id]);
        return rows;
    }

    async create(data) {
        const { evento_id, nombre, contacto } = data;
        const [result] = await db.query(
            'INSERT INTO asistentes (evento_id, nombre, contacto) VALUES (?, ?, ?)',
            [evento_id, nombre, contacto]
        );
        return result;
    }

    async update(id, data) {
        const { evento_id, nombre, contacto } = data;
        const [result] = await db.query(
            'UPDATE asistentes SET evento_id = ?, nombre = ?, contacto = ? WHERE id = ?',
            [evento_id, nombre, contacto, id]
        );
        return result;
    }

    async delete(id) {
        const [result] = await db.query('DELETE FROM asistentes WHERE id = ?', [id]);
        return result;
    }
}

module.exports = new AsistenteModel();

File: flask/models/asistente_model.py
"""
Modelo para la tabla Asistentes
"""
from config.database import Database

class AsistenteModel:
    """Maneja operaciones de base de datos para Asistentes"""
    
    @staticmethod
    def get_all():
        """Obtiene todos los asistentes con información del evento"""
        sql = """
        SELECT a.*, e.nombre as evento_nombre 
        FROM Asistentes a
        LEFT JOIN Eventos e ON a.evento_id = e.id
        ORDER BY a.nombre
        """
        return Database.execute_query(sql, fetch=True) or []
    
    @staticmethod
    def get_by_id(asistente_id):
        """Obtiene un asistente por su ID"""
        sql = "SELECT * FROM Asistentes WHERE id = %s"
        return Database.execute_query(sql, (asistente_id,), fetch_one=True)
    
    @staticmethod
    def create(nombre, contacto, evento_id):
        """Crea un nuevo asistente"""
        sql = "INSERT INTO Asistentes (nombre, contacto, evento_id) VALUES (%s, %s, %s)"
        return Database.execute_query(sql, (nombre, contacto, evento_id))
    
    @staticmethod
    def update(asistente_id, nombre, contacto, evento_id):
        """Actualiza un asistente existente"""
        sql = "UPDATE Asistentes SET nombre = %s, contacto = %s, evento_id = %s WHERE id = %s"
        return Database.execute_query(sql, (nombre, contacto, evento_id, asistente_id))
    
    @staticmethod
    def delete(asistente_id):
        """Elimina un asistente"""
        sql = "DELETE FROM Asistentes WHERE id = %s"
        return Database.execute_query(sql, (asistente_id,))